src/bootstrap/setup.rs

```diff
@@ -86,3 +88,41 @@ d) Install Rust from source"
     };
     Ok(template.to_owned())
 }
+
+// install a git hook to automatically run tidy --bless, if they want
+fn install_git_hook_maybe(src_path: &Path) -> io::Result<()> {
+    let mut input = String::new();
+    println!(
+        "Rust's CI will automatically fail if it doesn't pass `tidy`, the internal tool for ensuring code quality.
+If you'd like, x.py can install a git hook for you that will automatically run `tidy --bless` on each commit
+to ensure your code is up to par. If you decide later that this behavior is undesirable,
+simply delete the `pre-commit` file from .git/hooks."
+    );
+
+    let should_install = loop {
+        print!("Would you like to install the git hook?: [y/N] ");
+        io::stdout().flush()?;
+        io::stdin().read_line(&mut input)?;
+        break match input.trim().to_lowercase().as_str() {
+            "y" | "yes" => true,
+            // is this the right way to check for "entered nothing"?
+            "n" | "no" | "" => false,
+            _ => {
+                println!("error: unrecognized option '{}'", input.trim());
+                println!("note: press Ctrl+C to exit");
+                continue;
+            }
+        };
+    };
+
+    if should_install {
+        let src = src_path.join("/etc/pre-commit.rs");
```

Comment:
This will break on windows I think, it uses `\` for path separators.
```suggestion
        let src = src_path.join("etc").join("pre-commit.sh");
```